service_identity/src/main/kotlin/org/thechance/service_identity/api/endpoints/userRoutes.kt

```diff
@@ -0,0 +1,50 @@
+package org.thechance.service_identity.api.endpoints
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import org.koin.ktor.ext.inject
+import org.thechance.service_identity.api.model.UserDto
+import org.thechance.service_identity.domain.usecases.user.UserUseCaseContainer
+
+fun Route.userRoutes() {
+
+    val userUseCaseContainer: UserUseCaseContainer by inject()
+
+    route("/user") {
+
+        get("/{id}") {
+            val id = call.parameters["id"] ?: ""
+            val user = userUseCaseContainer.getUserByIdUseCase.invoke(id).toUserDto()
+            call.respond(HttpStatusCode.OK, user)
+        }
+
+        post {
+            val user = call.receive<UserDto>()
+            val result = userUseCaseContainer.createUserUseCase.invoke(user.toUser())
+            call.respond(HttpStatusCode.Created, result)
+        }
+
+        put("/{id}") {
+            val id = call.parameters["id"] ?: ""
+            val userDto = call.receive<UserDto>()
+            val result = userUseCaseContainer.updateUserUseCase.invoke(id, userDto.toUser())
+            call.respond(HttpStatusCode.Created, result)
+        }
+
+        delete("/{id}") {
+            val id = call.parameters["id"] ?: ""
+            val result = userUseCaseContainer.deleteUserUseCase.invoke(id)
+            call.respond(HttpStatusCode.OK, result)
+
+        }
+    }
+
+    get("/users") {
+        val users = userUseCaseContainer.getUsersUseCase.invoke()
+        call.respond(HttpStatusCode.OK, users.map { it.toUserDto() })
+    }
```

Comment:
You don't have to move this outside the route. Just change "user" to "users" for the route. Also, the other methods like post, put, and delete will be planned under the users route. It's supposed to look like this:

```
route("/users") {
   get{} // This will return all users.
   post{} // This will create a user inside the users collection.
   put("/{id}") // This will update a user inside the users collection.
   ...
}

```